src/ckb/network.rs

```diff
@@ -1431,36 +1392,31 @@ impl ServiceProtocol for Handle {
             context.proto_id, context.session.id
         );
 
-        if let Some(peer_id) = context.session.remote_pubkey.clone().map(PeerId::from) {
-            self.send_actor_message(NetworkActorMessage::new_event(
-                NetworkActorEvent::PeerDisconnected(peer_id, context.session.clone()),
-            ));
-        } else {
-            warn!(
-                "Peer disconnected without remote pubkey {:?}",
-                context.session
-            );
+        match context.session.remote_pubkey.as_ref() {
+            Some(pubkey) => {
+                let peer_id = PeerId::from_public_key(pubkey);
+                self.send_actor_message(NetworkActorMessage::new_event(
+                    NetworkActorEvent::PeerDisconnected(peer_id, context.session.clone()),
+                ));
+            }
+            None => {
+                unreachable!("Received message without remote pubkey");
```

Comment:
AFAIK, executing the code `unreachable!` will result in an undefined behavior. I don't know if it is possible for the adversary to fake a tentacle connection without providing any remote pubkey. If this is a legal operation for tentacle, then this will make the host panic.

Reply:
Then, this is fine for me.